contracts/src/Gateway.sol

```diff
@@ -157,78 +141,163 @@ contract Gateway is IGateway, IInitializable, IUpgradable {
             revert InvalidProof();
         }
 
+        bool success = dispatch(
+            message.command,
+            message.params,
+            maxDispatchGas
+        );
+
+        processReward(channel.agent, msg.sender, startGas, message.maxFeePerGas, message.reward);
+
+        emit IGateway.InboundMessageDispatched(
+            message.channelID,
+            message.nonce,
+            message.id,
+            success
+        );
+    }
+
+    /// @dev Submit a message from Polkadot for verification and dispatch
+    /// @param message A message produced by the OutboundQueue pallet on BridgeHub
+    /// @param leafProof A message proof used to verify that the message is in the merkle tree committed by the OutboundQueue pallet
+    /// @param headerProof A proof that the commitment is included in parachain header that was finalized by BEEFY.
+    function submitV2(
+        InboundMessageV2 calldata message,
+        bytes32[] calldata leafProof,
+        Verification.Proof calldata headerProof
```

Comment:
would it be more clear to rename `leafProof` to `messageProof` and `headerProof` to `mmrProof`?